.github/workflows/bench.yaml

```diff
@@ -0,0 +1,41 @@
+name: 📈 Benchmark
+
+on:
+  push:
+    branches: [ main ]
+
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Install Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          override: true
+          toolchain: stable
+          profile: minimal
+
+      - name: Cache Project
+        uses: Swatinem/rust-cache@v2
+
+      - name: Run benchmark
+        run: cargo bench -q -p wnfs -- --output-format bencher | tee output.txt
+
+      - name: Store Benchmark Result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: Rust Benchmark
+          tool: 'cargo'
+          output-file-path: output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
```

Comment:
@appcypher maybe we shouldn't auto push directly: https://github.com/benchmark-action/github-action-benchmark#run-only-on-your-branches, but do something like

```
auto-push: ${{ github.event_name == 'push' && github.repository == 'wnfs-wg/rs-wnfs' && github.ref == 'refs/heads/main' }}
```